### PR TITLE
add retries counter to ci

### PR DESCRIPTION
**Pull request by cmelone**
[proof of concept, don't merge]

In anticipation of future changes to CI, we'll need to know how many times a specific job has been retried. Unfortunately, this isn't available as a clean variable in gitlab (see issues [195618](https://gitlab.com/gitlab-org/gitlab/-/issues/195618)/[27589](https://gitlab.com/gitlab-org/gitlab/-/issues/27589)), so we use a [workaround](https://stackoverflow.com/questions/74548131/how-to-identify-retry-pipeline-jobs-in-gitlab) with artifacts.

Basically want to make sure this artifact/variable is available to the `pre_build_script` of the subsequent job.

### CODE REVIEW

**Comment by cmelone** [comment]
@kwryankrattiger what would be the best way to test if this works? edit: I'm testing it on a test repo on gitlab.spack.io

**Comment by cmelone** [comment]
artifacts are not easily shared between retries so I'm closing this
